design/secrets_provider_milestone1_design.md

```diff
@@ -42,54 +42,103 @@
+
+The following image shows the desicion flow chart that the customer will use to determine how to configure Secrets Provider depends on the value he wants to get:
+
+![Secrets Provider flavors desicion flow chart](https://user-images.githubusercontent.com/31179631/85747023-975bf500-b70f-11ea-8e26-1134068fe655.png)
 
-- Milestone 1: Serve multiple applications, deploy as as separate entity
-- Milestone 2: Handle rotation, updating K8s Secrets with Conjur secret value
-- Milestone 3: Support reading secrets from files
-- Milestone 4: Listen for K8s Secret creation
-- Milestone 5: Trigger K8s Secret update on Conjur secret value update
+This flow chart shows 6 different variations of Secrets Provider:
 
-For the purposes of this document, we will *only* be addressing the first Milestone.
+| Secrets Provider variation                            | Business value                                               | Implementation status |
+| ----------------------------------------------------- | ------------------------------------------------------------ | --------------------- |
+| Run as **Init Container** and sync to **K8s Secrets** | Supply secrets on application load, based on K8s secrets     | Implemented           |
+| Run as **Job** and sync to **K8s Secrets**            | Supply secrets once for many applications, based on K8s secrets | Milestone 1           |
+| Run as **Deployment** and sync to **K8s Secrets**     | Supply secrets for many applications and support rotation, based on K8s secrets | Future (Milestone 2)  |
+| Run as **Sidecar** and sync to **K8s Secrets**        | Support secrets rotation, based on K8s secrets               | Future (Milestone 2)  |
+| Run as **Init Container** and sync to **files**       | Supply secrets on application load, pushed as files to shared volume | Future (Milestone 3)  |
+| Run as **Sidecar** and sync to **files**              | Supply secrets for one application and support rotation, pushed as files to shared volume | Future (Milestone 3)  |
 
+All these variations will be supported using the same Secrets Provider image, and the behavior will vary dynamically depends on the chosen deployment.
 
+In this document we will show the detailed solution for **Milestone 1** only.
 
-### Milestone 1: Serve Multiple Applications
+### Milestone 1: Serve Multiple Applications once (no rotation)
 
 #### Design
 
-The Secrets Provider for K8s will no longer run as an *init* container attached to the application pod, rather run as a separate single Job in a namespace. A Job creates a Pod to perform a task and terminates upon completion. Once the Pod is spun up, it will authenticate to Conjur/DAP in the same way it does today, via authn-k8s. The Pod will then fetch the K8s secret denoted with a label and update them with the Conjur secrets they require.
+Run Secrets Provider as a **Job** using the same solution for init container.
```

Comment:
@abrahamko We are creating a _new_ feature and should not continuously be referencing a previous one
`Run Secrets Provider as a **Job**`